### PR TITLE
feat: `no-case-declarations` add suggestions

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[x] Add ~autofix~ suggestion to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

Fixes https://github.com/eslint/eslint/issues/18349

#### What changes did you make? (Give an overview)

Adds a suggestion fixer to `no-case-declarations` to add `{` `}` brackets around blocks.

Also adds a couple new tests to make sure switches with multiple violating cases report on the right case (consequent).

#### Is there anything you'd like reviewers to focus on?

I'd considered adding logic for friendlier `{` `}` bracket placement but then decided against the added complexity.

### CODE REVIEW

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

Leaving it open as, per our [policies](https://eslint.org/docs/latest/maintain/review-pull-requests), it requires at least one additional approval from any other team member (because this is a new feature), and also a waiting period of 2 days before merging.

**Comment by fasttime** [comment]
When a declared identifier is used in another `case` block, applying the suggestion can change the logic, for example:

```js
// ecmaVersion: "latest"
switch ("foo") {
    case "bar":
        function baz() { }
        break;
    default:
        baz();
}
```

compared to:

```js
// ecmaVersion: "latest"
switch ("foo") {
    case "bar":
        { function baz() { }
        break; }
    default:
        baz(); // TypeError!
}
```

I assume this is acceptable for a suggestion, but maybe we should add a unit test to show that we are aware of the risk and we know what we are doing?

**Review by fasttime — APPROVED** [review]
LGTM, thanks! Leaving open to respect the the 2 days period.

**Review by mdjermanovic — APPROVED** [review]

**Pull request merged**
